.github/workflows/deploy.yml

```
@@ -0,0 +1,24 @@
+name: deploy
+on:
+  release:
+    types: [published]
```

I think the trigger type should be `on: push:` with tags and the main branch.

This will allow snapshot releases to be published for each commit, and also published to maven when the tag is created: https://github.com/sbt/sbt-ci-release/blob/main/.github/workflows/release.yml

```suggestion
on:
  push:
    branches: [main]
    tags: ["*"]
```